### ISSUE TITLE
[BUG] HTML `<label>` elements are not referencing the item they are labeling

### ISSUE THREAD

**Issue by dasnessie** [bokeh/bokeh#13585]
### Software versions

Python version : 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
IPython version : (not installed)
Tornado version : 6.4
Bokeh version : 3.3.1
BokehJS static path : /home/dasnessie/Sandbox/bokeh-minimalexample/venv/lib/python3.11/site-packages/bokeh/server/static
node.js version : (not installed)
npm version : (not installed)
jupyter_bokeh version : (not installed)
Operating system : Linux-6.1.0-11-amd64-x86_64-with-glibc2.36

### Expected behavior

I created a bokeh FileInput. I expected it's label to reference the input field, as it is specified for HTML:

When creating a HTML [`<label>`-Element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label), it needs to either wrap around the item it is labeling like this:

```
<label>
  Enter your username:
  <input name="username" type="text" />
</label>
```

or reference the item it is labeling like this:

```
<label for="username">Enter your username:</label>
<input id="username" name="username" type="text" />
```

### Observed behavior

The `<label>` element does not reference the input field and as such is not related to the input field at all, not following the HTML specification and creating accessibility issues.

I have seen this behavior on different input fields, like `TextInput`, `FileInput`, and `NumericInput`.

The HTML that is served looks like this: 

```
<div class="bk-TextInput">
  <div class="bk-input-group">
    <label>Enter your username:</label>
    <div class="bk-input-container">
      <input type="text" class="bk-input" placeholder="">
    </div>
  </div>
</div>
```

### Example code

```Python
from bokeh.models import TextInput
from bokeh.plotting import curdoc

input = TextInput(name="inputField", title="Enter your username:")
curdoc().add_root(input)
```

**Comment by bryevdv** [comment]
@mattpap a new contributor on slack reached out looking for a good first issue. Can you comment on this one? It seems like it could be self-contained (e.g in [`input_widget.ts`](https://github.com/bokeh/bokeh/blob/branch-3.4/bokehjs/src/lib/models/widgets/input_widget.ts) and/or possibly [`core/dom.ts`](https://github.com/bokeh/bokeh/blob/branch-3.4/bokehjs/src/lib/core/dom.ts)?) 

@dasnessie the guidelines seem a little loose, would this be acceptable (i.e. with another `div` enclosing the `input` inside the `label`)?
```html
<div class="bk-TextInput">
  <div class="bk-input-group">
    <label>Enter your username:
      <div class="bk-input-container">
        <input type="text" class="bk-input" placeholder="">
      </div>
    </label>
  </div>
</div>
```
Otherwise, we don't assign `id` to elements anymore AFAIK so going that route would be a bigger lift.

**Comment by mattpap** [comment]
Rearranging DOM elements will likely cause CSS styling issues. However, using a simple ID, like `id="input"` and then `<label for="input">` is fine, because components are isolated from each other, so IDs don't have to be unique across the entire DOM tree.

**Comment by bryevdv** [comment]
> components are isolated from each other, so IDs don't have to be unique across the entire DOM tree.

Didn't realize that, that's great. This seems like a suitable new contributor issue in that case.

**Comment by ghamr** [comment]
Hello, I'm working with Dasha on this @dasha-moskv. We are new to TS so we aren't super familiar with how to modify the TS code to add the for="input" to the label element. We tried modifying line 162 of the code by doing this:
return label({for: {input}, style: {display}}, content, this.desc_el)

this:
return label({for: this.input_el.id, style: {display}}, content, this.desc_el);

and this:
return label({for: "input", style: {display}}, content, this.desc_el)

but none of the above worked. We also don't know where the id="input" fits into all of this. 
Any help is greatly appreciated.
